1566 quota

hyi opened the pull request:
@pkdash This PR includes all changes related to quota that are needed when working with @mjstealey for quota script integration as well as fixing display of validation errors on create resource page (which is a regression error that needs to be fixed). @mjstealey and I have gone through the whole workflow on alpha and it appears to work as expected, so I am creating this PR to get the code review started. I am creating this PR against develop branch for code review, but can create a corresponding PR against release 1.11 branch as needed when I get a +1. Please code review this PR when you have time. Thanks.

pkdash commented:
@hyi I assume this PR is targeted for release 1.12.

pkdash commented on hs_core/hydroshare/utils.py:
```diff
@@ -669,16 +669,16 @@ def convert_file_size_to_unit(size, unit):
                               'or TB')
 
-    kbsize = size / 1000.0
+    kbsize = size / 1024.0
```
May be use a variable for the value 1024.0

hyi commented:
@pkdash We'll decide whether it is targeted at 1.11 or 1.12 after tomorrow's demo based on feedback from group. Then I'll create a new PR against 1.11 if we decide to target this on 1.11

pkdash commented on hs_core/hydroshare/utils.py:
```diff
@@ -699,16 +699,18 @@ def validate_user_quota(user, size):
             hard_limit = qmsg.hard_limit_percent
-            used_size = round(uq.used_value + convert_file_size_to_unit(size, uq.unit))
-            used_percent = round(used_size*100.0/uq.allocated_value)
+            used_size = uq.used_value + convert_file_size_to_unit(size, uq.unit)
```
Consider making the followings as properties of UserQuota class:
```used_size```
```used_percent```
```rounded_percent```
```rounded_used_val``` - > ```rounded_used_size```

pkdash commented on theme/utils.py:
```diff
@@ -17,40 +17,43 @@ def get_quota_message(user):
     for uq in user.quotas.all():
-        percent = round(uq.used_value * 100.0 / uq.allocated_value)
-        if percent >= hard_limit or uq.remaining_grace_period == 0:
+        percent = uq.used_value * 100.0 / uq.allocated_value
```
Then you can use the properties of ```uq``` here instead.

pkdash commented:
@hyi I have left couple of comments.

hyi commented:
@pkdash I have addressed your code review comments. See if you can +1 now. @mjstealey Let me know if you want me to create a new PR against release 1.11 for this quota branch.

pkdash commented:
+1

hyi commented:
@mjstealey I merged this PR to develop and left the other corresponding PR against release 1.11 open since I thought it will certainly go into release 1.12. Then on second thought, it might not be as easy as I thought meaning if quota does need to go into release 1.11 by merging the other corresponding PR, when you merge release 1.11 back to develop later, merging this into develop first may pose conflict and give you issues. If this is indeed true, I can easily revert merging back. Let me know whatever you think I should do that will work best for your upcoming deployment. Thanks.

mjstealey commented:
@hyi - Given all the discussion around quota during the team call, we'd want the ability to "mute" the quota popups and emails if we do put it into 1.11.0. 

I don't see the harm in the code being in 1.11.0, so long as we can prevent it from being activated if need be. From the backend that is as simple as not calling the **update_used_quota** command. But not sure what work would need to be on the front end to enable an on/off switch...

hyi commented:
@mjstealey The "muted" version of quota implementation is already in our code base and deployment a couple of releases back. This quota PR introduces additional changes made during our quota integration testing while enabling previously muted quota messages, so it is aimed at full quota deployment with your script wired in. If you think it is better to delay quota full implementation to release 1.12, then I did the right thing by merging that PR to develop to get ready for release 1.12, and I can delete the other corresponding PR I created against release 1.11. Let me know what you think, and if you agree, I will then proceed to delete the PR created against release 1.11, and we can assume quota full implementation will be ready for release 1.12 since the code has already been merged into develop.

mjstealey commented:
@hyi - I'd lean towards putting it into 1.11.0 then.

> The "muted" version of quota implementation is already in our code base and deployment a couple of releases back.

hyi commented:
@mjstealey Just to confirm, you do mean you lean towards putting it into **1.11** (not **1.12**), right? If it is 1.11, should I revert the merging action for the PR against develop, but merging the other PR into release 1.11 branch instead? If it is 1.12, then I just need to delete the other PR against release 1.11. Want to confirm with you just to be sure before I take any further action.
